src/api/objects/ObjectAPI.js

```diff
@@ -189,6 +189,16 @@ define([
         });
     };
 
+    ObjectAPI.prototype.search = function* (options) {
+        for (const provider of Object.values(this.providers)) {
+            if (provider.search !== undefined) {
+                yield provider.search(options);
+            }
+        }
+
+        yield this.fallbackProvider.superSecretFallbackSearch(options);
```

Comment:
We should convert these to new style domain objects so we're not handling a mix of legacy and new style domain objects in the tree

Reply:
good call. done